modules/apps/click-to-chat/click-to-chat-web/src/main/java/com/liferay/click/to/chat/web/internal/servlet/taglib/ClickToChatTopHeadJSPDynamicInclude.java

```diff
@@ -67,137 +66,149 @@ public void include(
+				String clickToChatEnabled =
+					typeSettingsUnicodeProperties.getProperty(
+						"clickToChatEnabled");
 
-			if (clickToChatEnabled == null) {
-				if (!_clickToChatConfiguration.enabled()) {
-					return;
+				if (clickToChatEnabled == null) {
+					if (!clickToChatConfiguration.enabled()) {
+						return;
+					}
 				}
-			}
-			else {
-				if (!GetterUtil.getBoolean(clickToChatEnabled)) {
-					return;
+				else {
+					if (!GetterUtil.getBoolean(clickToChatEnabled)) {
+						return;
+					}
 				}
-			}
 
-			String clickToChatGuestUsersAllowed =
-				typeSettingsUnicodeProperties.getProperty(
-					"clickToChatGuestUsersAllowed");
+				String clickToChatGuestUsersAllowed =
+					typeSettingsUnicodeProperties.getProperty(
+						"clickToChatGuestUsersAllowed");
 
-			if (clickToChatGuestUsersAllowed == null) {
-				if (!_clickToChatConfiguration.guestUsersAllowed() &&
-					!themeDisplay.isSignedIn()) {
+				if (clickToChatGuestUsersAllowed == null) {
+					if (!clickToChatConfiguration.guestUsersAllowed() &&
+						!themeDisplay.isSignedIn()) {
 
-					return;
+						return;
+					}
 				}
-			}
-			else {
-				if (!GetterUtil.getBoolean(clickToChatGuestUsersAllowed) &&
-					!themeDisplay.isSignedIn()) {
+				else {
+					if (!GetterUtil.getBoolean(clickToChatGuestUsersAllowed) &&
+						!themeDisplay.isSignedIn()) {
 
-					return;
+						return;
+					}
 				}
-			}
 
-			String clickToChatProviderAccountId =
-				typeSettingsUnicodeProperties.getProperty(
-					"clickToChatProviderAccountId",
-					_clickToChatConfiguration.chatProviderAccountId());
-			String clickToChatProviderId =
-				typeSettingsUnicodeProperties.getProperty(
-					"clickToChatProviderId",
-					_clickToChatConfiguration.chatProviderId());
+				String clickToChatProviderAccountId =
+					typeSettingsUnicodeProperties.getProperty(
+						"clickToChatProviderAccountId",
+						clickToChatConfiguration.chatProviderAccountId());
+				String clickToChatProviderId =
+					typeSettingsUnicodeProperties.getProperty(
+						"clickToChatProviderId",
+						clickToChatConfiguration.chatProviderId());
 
-			if (Validator.isNull(clickToChatProviderAccountId) ||
-				Validator.isNull(clickToChatProviderId)) {
+				if (Validator.isNull(clickToChatProviderAccountId) ||
+					Validator.isNull(clickToChatProviderId)) {
 
-				return;
-			}
+					return;
+				}
 
-			httpServletRequest.setAttribute(
-				ClickToChatWebKeys.CLICK_TO_CHAT_PROVIDER_ACCOUNT_ID,
-				clickToChatProviderAccountId);
-			httpServletRequest.setAttribute(
-				ClickToChatWebKeys.CLICK_TO_CHAT_PROVIDER_ID,
-				clickToChatProviderId);
+				httpServletRequest.setAttribute(
+					ClickToChatWebKeys.CLICK_TO_CHAT_PROVIDER_ACCOUNT_ID,
+					clickToChatProviderAccountId);
+				httpServletRequest.setAttribute(
+					ClickToChatWebKeys.CLICK_TO_CHAT_PROVIDER_ID,
+					clickToChatProviderId);
+			}
+		}
+		catch (PortalException portalException) {
+			throw new RuntimeException(portalException);
```

Comment:
Should we just log this exception? :thinking:

Reply:
@alejoceballos , I think is better throw new SystemException(portalException) instead RuntimeException;
You can do this:
```suggestion
if (_log.isDebugEnabled()) {
     _log.debug(portalException, portalException);
}
 throw new SystemException(portalException);
```